packages/cloud_functions/cloud_functions_platform_interface/lib/cloud_functions_platform_interface.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library cloud_functions_platform_interface;
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show required, visibleForTesting;
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+part 'src/method_channel_cloud_functions.dart';
+
+/// The interface that implementations of `cloud_functions` must extend.
+///
+/// Platform implementations should extend this class rather than implement it
+/// as `cloud_functions` does not consider newly added methods to be breaking
+/// changes. Extending this class (using `extends`) ensures that the subclass
+/// will get the default implementation, while platform implementations that
+/// `implements` this interface will be broken by newly added
+/// [CloudFunctionsPlatform] methods.
+abstract class CloudFunctionsPlatform extends PlatformInterface {
+  static final Object _token = Object();
+
+  /// Constructs a CloudFunctionsPlatform
+  CloudFunctionsPlatform() : super(token: _token);
+
+  /// The default instance of [CloudFunctionsPlatform] to use.
+  ///
+  /// Platform-specific plugins should override this with their own class
+  /// that extends [CloudFunctionsPlatform] when they register themselves.
+  ///
+  /// Defaults to [MethodChannelCloudFunctions].
+  static CloudFunctionsPlatform get instance => _instance;
+
+  static CloudFunctionsPlatform _instance = MethodChannelCloudFunctions();
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [CloudFunctionsPlatform] when they register themselves.
+  // TODO(amirh): Extract common platform interface logic.
+  // https://github.com/flutter/flutter/issues/43368
+  static set instance(CloudFunctionsPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  /// Invokes the specified cloud function.
+  ///
+  /// The data passed into the cloud function can be any of the following types:
+  ///
+  /// `null`
+  /// `String`
+  /// `num`
+  /// [List], where the contained objects are also one of these types.
+  /// [Map], where the values are also one of these types.
+  ///
+  /// @param appName name of the Firebase application to which the function belongs
```

Comment:
we don't use the JavaDoc style with `@param` sections for each parameter. The common style in Dartdocs is to write a paragraph (or paragraphs) describing the parameters and wrapping the parameter name in square brackets so they are linked. For example:

```
/// The [appName] is the name of the Firebase app to which the function belongs. ...
```